runtime/v2/runc/v1/service.go

```diff
@@ -147,6 +147,12 @@ func (s *service) StartShim(ctx context.Context, id, containerdBinary, container
 			return "", err
 		}
 	}
+	defer func() {
```

Comment:
In case or returning error, `cmd.Process.Kill()` will execute first and then socket close. Is this order correct?